consensus/src/block_storage/execution_pool/block_window_test.rs

```diff
@@ -0,0 +1,588 @@
+
+    // Getting the ordered block window for a genesis block should return an empty window
+    // | blocks inserted | window_size | round | ordered_block_window block count |
+    // |-----------------|-------------|-------|----------------------------------|
+    // | 0               | 3           | 0     | 0                                |
+    let block = curr_pipelined_block.block();
+    let blocks = get_blocks_from_block_store_and_window(block_store.clone(), block, window_size);
+    assert_eq!(blocks.len(), 0);
+    assert_eq!(round, 0);
+
+    // | blocks inserted | window_size | round | ordered_block_window block count |
+    // |-----------------|-------------|-------|----------------------------------|
+    // | 1               | 3           | 1     | 1                                |
+    round += 1;
+    curr_pipelined_block = inserter
+        .insert_block(&curr_pipelined_block, round, None)
+        .await;
+    let block = curr_pipelined_block.block();
+    let blocks = get_blocks_from_block_store_and_window(block_store.clone(), block, window_size);
+
+    assert_eq!(blocks.len(), 1);
+    assert_eq!(round, 1);
+
+    // | blocks inserted | window_size | round | ordered_block_window block count |
+    // |-----------------|-------------|-------|----------------------------------|
+    // | 2               | 3           | 2     | 2                                |
+    round += 1;
+    curr_pipelined_block = inserter
+        .insert_block(&curr_pipelined_block, round, None)
+        .await;
+    let blocks = get_blocks_from_block_store_and_window(
+        block_store.clone(),
+        curr_pipelined_block.block(),
+        window_size,
+    );
+    assert_eq!(blocks.len(), 2);
+    assert_eq!(blocks.first().unwrap().round(), 0);
+    assert_eq!(round, 2);
+
+    // | blocks inserted | window_size | round | ordered_block_window block count |
+    // |-----------------|-------------|-------|----------------------------------|
+    // | 3               | 3           | 3     | 2                                |
+    round += 1;
+    curr_pipelined_block = inserter
+        .insert_block(&curr_pipelined_block, round, None)
+        .await;
+    let blocks = get_blocks_from_block_store_and_window(
+        block_store.clone(),
+        curr_pipelined_block.block(),
+        window_size,
+    );
+    assert_eq!(blocks.len(), 2);
+    assert_eq!(blocks.first().unwrap().round(), 1);
+    assert_eq!(blocks.get(1).unwrap().round(), 2);
+    assert_eq!(round, 3);
+
+    // | blocks inserted | window_size | round | ordered_block_window block count |
+    // |-----------------|-------------|-------|----------------------------------|
+    // | 4               | 3           | 4     | 2                                |
+    round += 1;
+    curr_pipelined_block = inserter
+        .insert_block(&curr_pipelined_block, round, None)
+        .await;
+    let blocks = get_blocks_from_block_store_and_window(
```

Comment:
these are a lot of duplicate code, seems just need a for loop for the basic cases?

Reply:
I think i might delete this test actually. I'm not sure it adds much value...